File: api/src/endpoint/question/get_question.rs
use serde::{Deserialize, Serialize};
use api_derive::EndpointRequest;

use crate::{endpoint::Endpoint, resource::question::QuestionResource};

pub struct GetQuestion;

impl Endpoint for GetQuestion {
    const PATH: &'static str = "/repository/{owner}/{repo}/question/{number}";
    const METHOD: http::Method = http::Method::GET;

    type Request = GetQuestionRequest;
    type Response = GetQuestionResponse;
}

#[derive(EndpointRequest, Debug, Serialize, Deserialize)]
pub struct GetQuestionRequest;

pub type GetQuestionResponse = QuestionResource;

File: gitdot-api-ts/src/endpoint/question/get-question.ts
import { z } from "zod";

import { QuestionResource } from "../../resource";

export const GetQuestionRequest = z.object({});
export type GetQuestionRequest = z.infer<typeof GetQuestionRequest>;

export const GetQuestion = {
  path: "/repository/{owner}/{repo}/question/{number}",
  method: "GET",
  request: GetQuestionRequest,
  response: QuestionResource,
} as const;
export type GetQuestion = typeof GetQuestion;
